### PR TITLE
Deprecate SDIRK time integrators

### PR DESCRIPTION
**Pull request by blaisb**
# Description of the problem

- The SDIRK time integrators were poorly maintained and gave a false illusion that they were actually useable in contexts in which they were not.
-  No one was actually using the time integrators and they have worse performance than the equivalent BDF one

# Description of the solution

- Remove the SDIRK integrator. If we actually need them in the future, we can do a better reimplementation.

# Documentation

- Updated documentation

### CODE REVIEW

File: include/core/sdirk.h
Comment (oguevremont) [inline include/core/sdirk.h:61]: Are we keeping some lines, just in case?

File: source/core/parameters.cc
Comment (oguevremont) [inline source/core/parameters.cc:173]: Sdirk options are still in the declare_parameters explanatory text, they should be removed

File: source/fem-dem/gls_sharp_navier_stokes.cc
Comment (oguevremont) [inline source/fem-dem/gls_sharp_navier_stokes.cc:1]: There is still #include <core/sdirk.h>, maybe it needs to be removed. It is also in other files

File: include/solvers/navier_stokes_base.h
Comment (lpsaavedra) [inline include/solvers/navier_stokes_base.h:447]: I think all the multiphysics solvers have this parameter as well and needs to be deleted? Cahn Hilliard, VOF, heat transfer, tracer?

Comment (lpsaavedra) [inline include/solvers/navier_stokes_base.h:447]: I think all the multiphysics (heat transfer, tracer, vof and cahn hilliard) and it should be removed as well?

File: source/core/parameters.cc
Comment (lpsaavedra) [inline source/core/parameters.cc:1]: In this file, sdirk options need to be removed when declaring the parameters of simulation control (between lines 61 and 70)...

**Review by oguevremont — COMMENTED** [review]
Good! Some questions/comments, but nothing major

File: source/solvers/cahn_hilliard_scratch_data.cc
Comment (lpsaavedra) [inline source/solvers/cahn_hilliard_scratch_data.cc:1]: Related to my comment above: `cahn_hilliard_scratch_data.h` still uses solution_stages in several places

File: source/solvers/heat_transfer_scratch_data.cc
Comment (lpsaavedra) [inline source/solvers/heat_transfer_scratch_data.cc:1]: Related to my comment above: `heat_transfer_scratch_data.h` still uses solution_stages in several places

File: source/solvers/tracer_scratch_data.cc
Comment (lpsaavedra) [inline source/solvers/tracer_scratch_data.cc:1]: Related to my comment above: `tracer_scratch_data.h` still uses solution_stages in several places

File: source/solvers/vof_scratch_data.cc
Comment (lpsaavedra) [inline source/solvers/vof_scratch_data.cc:1]: Related to my comment above: vof_scratch_data.h still uses solution_stages in several places

File: include/core/sdirk.h
Comment (lpsaavedra) [inline include/core/sdirk.h:1]: Do we still want to have `sdirk.h` and `sdirk.cc` in `source/core/CMakeLists.txt`? We could remove it if it is not being used. We should also avoid including the file if not used (remove the `#include <core/sdirk.h>` from all files) in source solvers.

**Review by lpsaavedra — CHANGES_REQUESTED** [review]
Some questions/comments need to be addressed before merging. Also, there are two places in the documentation where sdirk is still mentioned and this should be updated:

- [x] `/doc/source/examples/incompressible-flow/2d-transient-flow-around-cylinder/2d-transient-flow-around-cylinder.rst` line 304.
- [ ] `/doc/source/examples/sharp-immersed-boundary-solver/sedimentation-1-particle/sedimentation-1-particle.rst` line 64.

File: include/core/sdirk.h
Comment (blaisb) [reply include/core/sdirk.h:1]: Yeah I'd rather leave the file for the coefficients, but not use it anymore. I'll clean that up.

**Review by lpsaavedra — APPROVED** [review]
Seems good to me! Only two minor additional changes: the #include directive should be removed also from: `vans_assemblers.cc` and `gls_sharp_navier_stokes.cc`.

**Comment by blaisb** [comment]
> Seems good to me! Only two minor additional changes: the #include directive should be removed also from: `vans_assemblers.cc` and `gls_sharp_navier_stokes.cc`.

Done! I'll merge after CI :)

**Review by oguevremont — APPROVED** [review]

**Pull request merged**
